### ISSUE TITLE
prompting for a password

### ISSUE THREAD

**Issue by slisznia** [casey/just#374]
My recipe needs to deploy to prod and take a password silently. Without using ENV vars, is it possible to prompt and capture a password into a just-variable? I don't believe "read" command is useful here since it takes a bash variable to set the result into.

i.e.
PASS = \`prompt without echoing and return\`

**Comment by casey** [comment]
I don't think there's a good way to do this with just variables.

However, you can do this with env variables in a shebang recipe:

```
pw:
	#!/usr/bin/env bash
	echo Please enter your password:
	read -s pw
	echo $pw
```

Shebang recipes are executed as scripts, so variables work. You could also create a `.env` file containing the variables. Does that help?

**Comment by slisznia** [comment]
I saw the inline invocation of other languages through shebangs but how do you make the value of $pw (from your example) available outside of the shebang-recipe? The password needs to be reused by possibly many other recipes, so ideally it's stored in a variable.

**Comment by casey** [comment]
Ah, I see. Yeah, that's tough. You could read the password and then re-invoke just:

```
pw:
	#!/usr/bin/env bash
	echo Please enter your password:
	read -s pw
	echo $pw
	just pw=$pw deploy
```

Does putting it in a `.env` file work?

**Comment by slisznia** [comment]
No it doesn't - I tried that before posting this "issue" in the first place. I'm seeing "error: Backtick failed with exit code 1"

Neither does wrapping the password capture in a bash-helper-program like the prompt_for_password.sh  work. I'm out of ideas for now...

**Comment by casey** [comment]
Hmm, that's unfortunate. One possibility is that backtick commands aren't able to read from stdin, although I'm not sure why that would be the case.

**Comment by slisznia** [comment]
That seems to be the case exactly - it doesn't like the stdin program inside the backticks.

**Comment by casey** [comment]
I opened #376 to track that issue, which is definitely a bug.

**Comment by casey** [comment]
Fixed in #409!
